config/configuration_service.proto

```diff
@@ -7,65 +7,128 @@ package config;
-    rpc StartProcessingRequests(CommandRequest) returns (Response) {}
-}
+    rpc StartProcessingRequests (CommandRequest) returns (Response) {
+    }
+
+    //To Determine if the Daemon is processing requests are not.
+    // ("__Is_Daemon_Processing_Requests", "block_number",authentication_address) should be sent in the signature
+    rpc IsDaemonProcessingRequests (ReadRequest) returns (Response) {
+    }
+
 
 
+}
+
 //Used when you just need to read the current configuration
 message ReadRequest {
     //Signature will compromise of the below
     // ("_Request_Read", "block_number",authentication_address)
+    //OR
+    //("__Is_Daemon_Processing_Requests", "block_number",authentication_address)
     bytes signature = 1;
 
+    //current block number (signature will be valid only for short time around this block number)
+    uint64 current_block = 2;
+
+    //The address of the person who has rights to make configuration changes on Daemon
+    string user_address = 4;
+
 }
 
+
+
 //Used when you want to update the existing configuration
 message UpdateRequest {
     //Signature will compromise of the below
-    // ("_Request_Update", "block_number","updated_json_data",authentication_address)
+    // ("_Request_Update", "block_number",authentication_address)
     bytes signature = 1;
-    //Holds the updated configuration in the form of JSON,only the fields to be updated should be sent
-    string updated_json_data = 2;
+    //Indicates the updated configuration ( only the modified leaf and its changed value is passed ) Example (log.output.rotation_time_in_sec is a valid key )
+    map<string, string> updated_configuration = 2;
 
-}
+    //current block number (signature will be valid only for short time around this block number)
+    uint64 current_block = 3;
+
+    //The address of the person who has rights to make configuration changes on Daemon
+    string user_address = 4;
 
+}
 
 message CommandRequest {
     //Signature will compromise one of the below
     // ("_Request_Start", "block_number",authentication_address) for the StartProcessingRequests
     // ("_Request_Stop", "block_number",authentication_address) for the StopProcessingRequests
     bytes signature = 1;
+    //current block number (signature will be valid only for short time around this block number)
+    uint64 current_block = 2;
+    //The address of the person who has rights to make configuration changes on Daemon
+    string user_address = 3;
 }
 
 
-
 message Response {
     //Signature will compromise of the below
     //(block_number,authentication_address)
     bytes signature = 1;
+    //Indicates the current processing status ( true means Daemon is processing and false means it is not processing any requests)
+    bool current_processing_status = 2;
 }
 
 
+//Represents the configuration of current Daemon, please note every leaf level will be filled with these details.
+message ConfigurationParameter {
```

Comment:
We could return configuration scheme and configuration separately. In such case configuration is:
```proto
    //Indicates the configuration
    map<string, string> configuration = 2;
```
And scheme is array of `ConfigurationParameter(Scheme)` without value field.

Reply:
Thanks @vsbogd , with this approach , the client will have to keep track of key,value 
and key , Configuration and map them ..
Any additional advantages with the approach you propose ?

Reply:
Well, I thought about separating configuration scheme from configuration values mainly because of symmetry. When API has two methods `GetConfiguration` and `UpdateConfiguration` it is natural to be able passing results of `GetConfiguration` to `UpdateConfiguration`. I saw that in our API is not a case because `GetConfiguration` returns values coupled with scheme and I thought that decoupling them will make API more clear and natural. 

Another point: configuration scheme is bigger than configuration values but doesn't be changed frequently. So introducing separate method to get configuration scheme should decrease size of the data which is sent from daemon to client when `GetConfiguration` is called.

On the other hand adding separate method to get scheme has disadvantage as well. With separate method client need a way to find out when configuration scheme is changed. So at the end I think it is worth to add message to represent configuration scheme but still return it in `ConfigurationResponse`:

```protobuf
//Response sent back by Daemon on Read or Successful update
message ConfigurationResponse {
    //Signature will compromise of the below
    //(block_number,authentication_address,configuration)
    bytes signature = 1;

    //Holds the current configuration scheme of Daemon
    ConfigurationScheme scheme = 2;

    //Configuration (key, value) map
    map<string, string> configuration = 3;
}
```

Reply:
thanks @vsbogd , will do this